.github/workflows/main.yml

```diff
@@ -5,12 +5,67 @@ on:
   pull_request:
 
 jobs:
+  code_checks:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        code-check: [black, flakehell]
+        python-version: [3.7]
+    steps:
+      - uses: actions/checkout@v1
+
+      - uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+          architecture: 'x64'
+
+      - uses: actions/cache@v2
+        if: startsWith(runner.os, 'Linux')
+        with:
+          path: ${{ env.pythonLocation }}
+          key: ${{ env.pythonLocation }}-${{ runner.os }}-${{ matrix.python-version }}-${{ hashFiles('**/*requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ matrix.python-version }}-pip-
+
+      - name: Extract branch name  # works for both 'push' and 'pull_request' events
+        shell: bash
+        run: echo "##[set-output name=branch;]$( if [ ${{ github.event_name }} = push ]; then echo ${GITHUB_REF#refs/heads/}; else echo ${{ github.event.pull_request.head.ref }}; fi )"
+        id: extract_branch
+
+      - name: Install test requirements
+        run: |
+          pip install --upgrade pip
+          pip install --upgrade pip-tools
+          pip install -r requirements/test_requirements.txt
+
+      - name: Run code quality checks
+        run: |
+          FILES_TO_CHECK=$(git diff origin/master..origin/${{ steps.extract_branch.outputs.branch }} --name-only --diff-filter=ACM '*py')
```

Comment:
Pull request could be created to another branch except for `master`. Also, it would not work with `push` into the branch which will not be merged into master.

Reply:
Hm, good point. I'll see how to make `origin/master` dynamical.

About `push` - what do you mean? Diff will compare `target` branch with the current one and get the `diff`. Even if it `push` or `pull_request`.